src/components/Table/TableBase.tsx

```diff
@@ -557,36 +570,37 @@ export class TableBase<T extends {}> extends React.Component<
 			getRowClass,
 			className,
 			fuzzyPager,
+			pagination,
 		} = this.props;
 
 		const { page, sort } = this.state;
+		const serverSide = pagination?.serverSide;
 		const items = data || [];
-		const totalItems = items.length;
-		const _itemsPerPage = itemsPerPage || 50;
+		const totalItems = serverSide ? pagination?.totalItems : items.length;
+		const itemsPerPage = pagination?.itemsPerPage || DEFAULT_ITEMS_PER_PAGE;
```

Comment:
Given that this is a minor & `itemsPerPage` is still in the properties, we should
```suggestion
		const itemsPerPage = pagination?.itemsPerPage ?? this.props.itemsPerPage ?? DEFAULT_ITEMS_PER_PAGE;
```
both here and in `setPage`.
Otherwise we can do a major and remove it completely.
I think atm I'm in favor of the minor so that we can do a single major after the product call on how selection should work.